components/ws-manager-mk2/controllers/subscriber_controller.go

```diff
@@ -76,5 +80,21 @@ func (r *SubscriberReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 		}
 	}()
 
-	return c.Watch(source.Kind(mgr.GetCache(), &workspacev1.Workspace{}), &handler.EnqueueRequestForObject{})
+	filterByStatus := predicate.Funcs{
+		CreateFunc: func(ce event.CreateEvent) bool {
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			old := e.ObjectOld.(*workspacev1.Workspace)
+			new := e.ObjectNew.(*workspacev1.Workspace)
+
+			if !cmp.Equal(old.Spec.Ports, new.Spec.Ports) {
+				return true
+			}
+
+			return !equality.Semantic.DeepDerivative(old.Status, new.Status)
```

Comment:
should we use DeepEqual instead? As far as i understand DeepDerivative, it would e.g. ignore a status update where the snapshot url gets set (where previously it was an empty string). This is important for prebuilds, where bridge expects a snapshot url before the workspace moves to stopped (from [here](https://github.com/gitpod-io/gitpod/blob/2b907a60fdd450e81dc9440029d4da52aea92304/components/ws-daemon/pkg/controller/workspace_controller.go#L271C14-L271C14))

Reply:
Changed